### ISSUE TITLE
[Bug]: OPCache doesn't refresh when code changes

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/server#39079]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

OPCache should refresh when Nextcloud code changes. E.g. when an app is updated, or when the server is updated.

It appears that this mechanism doesn't work reliably for all instances. There is an increasing number of reports in the last few days.

Nextcloud does not have influence on OPCache. It is a PHP setting. Yet I'd like to collect some input and data to find out when this happens, if it needs to reported to PHP, a distro or a hoster, or the recommended OPCache settings are incorrect.

### Steps to reproduce

1. Run one version of Nextcloud
2. Replace code for a new version

### Expected behavior

New code is executed by PHP

### Installation method

None

### Nextcloud Server version

27

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Additional info

* https://github.com/nextcloud/polls/issues/2963

**Comment by ChristophWurst** [comment]
@juergen852 @roberto-thy @tobiasKaminsky could you please post your opcache settings? They should be in php.ini or a sibiling file.

**Comment by roberto-thy** [comment]
@ChristophWurst:
```
opcache.memory_consumption=512
opcache.interned_strings_buffer=10
opcache.max_accelerated_files=10000
opcache.validate_timestamps=0
opcache.revalidate_freq=60
opcache.save_comments=1
```

**Comment by ChristophWurst** [comment]
Thank you @roberto-thy. In your case it is expected to have to restart the web server for every change

> To disable the revalidation completely:
> 
> opcache.validate_timestamps = 0
> 
> Any change to config.php will then require either restarting PHP, manually clearing the cache, or invalidating this particular script.

https://docs.nextcloud.com/server/stable/admin_manual/installation/server_tuning.html#enable-php-opcache

**Comment by SystemKeeper** [comment]
Just wondering, if this is indeed disabled, wouldn’t that always be an issue when doing an update via the web updater ?

**Comment by roberto-thy** [comment]
I never consciously noticed any similar behavior before.

**Comment by juergen852** [comment]
> @juergen852 @roberto-thy @tobiasKaminsky could you please post your opcache settings? They should be in php.ini or a sibiling file.
We have no entries with opcache.xxx in config.php
`locate config.php | xargs cat | grep opcache` gives nothing.

**Comment by solracsf** [comment]
_Related:_ it would be great to have this switch for **apps** too:
https://github.com/nextcloud/server/blob/83eb6c3205997e32fcc1f276d71c2575d6e27bba/config/config.sample.php#L2160
So every update/upgrade should be performed on the CLI.

**Comment by tobiasKaminsky** [comment]
```
/etc grep opcache * -ir | grep -v ";"

php/php-fpm.conf_:zend_extension=opcache
php/php.ini:zend_extension=opcache
php/php.ini:[opcache]
php/php.ini:opcache.enable=1
php/php.ini:opcache.enable_cli=1
php/php.ini:opcache.memory_consumption=256
php/php.ini:opcache.interned_strings_buffer=32
php/php.ini:opcache.max_accelerated_files=10000
php/php.ini:opcache.revalidate_freq=1
php/php.ini:opcache.save_comments=1
```

**Comment by kesselb** [comment]
Further reports:

https://github.com/nextcloud/server/issues/37557
https://github.com/nextcloud/server/issues/38872
https://help.nextcloud.com/t/cli-update-but-http-not-updated-without-apache-restarted/165226

**Comment by ChristophWurst** [comment]
Fixed by https://github.com/nextcloud/server/pull/40349

**Comment by ChristophWurst** [comment]
> Fixed by #40349

At least for changes to the autoloader.

There are still reports that other code changes don't reflect neither.
